microsoft/testsuites/dpdk/dpdktestpmd.py

```diff
@@ -2,47 +2,310 @@
+        # clone devname repository into dpdk/examples
+        # allows us to build the small port identifier app
+        git.clone(
+            self._dpdk_devname_repo,
+            cwd=self._source_path.joinpath("examples"),
+            fail_on_exists=True,
+        )
+
+    def get_installed_version(self) -> VersionInfo:
+        return self._node.tools[Pkgconfig].get_package_version(
+            "libdpdk", update_cached=True
+        )
+
+    def _run_installation(self) -> None:
+        # check out devname into dpdk/examples (don't need to save the dir)
+        if self._sample_applications:
+            sample_apps = f"-Dexamples={','.join(self._sample_applications)}"
+        else:
+            sample_apps = ""
+        node = self._node
+        #
+        self.dpdk_build_path = node.tools[Meson].setup(
+            args=sample_apps, build_dir="build", cwd=self._source_path
+        )
+        node.execute(
+            "ninja",
+            cwd=self.dpdk_build_path,
+            timeout=1800,
+            expected_exit_code=0,
+            expected_exit_code_failure_message=(
+                "ninja build for dpdk failed. check build spew for missing headers "
+                "or dependencies. Also check that this ninja version requirement "
+                "has not changed for dpdk."
+            ),
+        )
+        node.execute(
+            "ninja install",
+            cwd=self.dpdk_build_path,
+            sudo=True,
+            expected_exit_code=0,
+            expected_exit_code_failure_message=(
+                "ninja install failed for dpdk binaries."
+            ),
+        )
+        node.execute(
+            "ldconfig",
+            cwd=self.dpdk_build_path,
+            sudo=True,
+            expected_exit_code=0,
+            expected_exit_code_failure_message="ldconfig failed, check for error spew.",
+        )
+        library_bashrc_lines = [
+            "export PKG_CONFIG_PATH=${PKG_CONFIG_PATH}:/usr/local/lib64/pkgconfig/",
+            "export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/usr/local/lib64/",
+        ]
+        node.tools[Echo].write_to_file(
+            ";".join(library_bashrc_lines),
+            node.get_pure_path("$HOME/.bashrc"),
+            append=True,
+        )
+        node.tools[Cp].copy(
+            self.dpdk_build_path.joinpath("examples/dpdk-devname"),
+            dest=node.get_pure_path("/usr/local/bin"),
+            sudo=True,
+        )
+
+
+class DpdkGitInstall(DpdkSourceInstall, GitInstall):
+    def __init__(
+        self,
+        node: Node,
+        os_dependencies: InstallDependencies,
+        git_repo: str,
+        git_ref: str,
+    ) -> None:
+        super().__init__(node, os_dependencies, git_repo, git_ref)
+
+    def _configure_installation(self) -> None:
+        if not self.git_ref:
+            git = self._node.tools[Git]
+            self.git_ref = git.get_tag(
+                self._source_path, filter_=r"^v.*"  # starts w 'v'
```

Comment:
It's the only deifference with the `DpdkSourceInstall`. The name of classes are confusing.

Reply:
We don't need to do this git ref filtering for other projects, it's project specific.
>It's the only deifference with the DpdkSourceInstall. 

Exactly. I only need to overload one or two functions per class, that's expected. There is a lot of shared code between the builds.